tests/__fixtures__/Rules/newline_before_scenario/fixture.js

```diff
@@ -0,0 +1,559 @@
+        ],
+        [
+            "without Rule: Scenario Outline",
+            `Feature: a feature file
+  Scenario Outline: a scenario outline`,
+            [generateProblem({ line: 2, column: 3 }, 0)],
+        ],
+        [
+            "without Rule: Background - Scenario Outline",
+            `Feature: a feature file
+  Background: a background
+  Scenario Outline: a scenario`,
+            [generateProblem({ line: 3, column: 3 }, 0)],
+        ],
+        [
+            "without Rule: multiple scenarios",
+            `Feature: a feature file
+  Background: a background
+
+
+  Scenario: a scenario
+
+
+
+  Scenario Outline: a scenario outline`,
+            [
+                generateProblem({ line: 5, column: 3 }, 2),
+                generateProblem({ line: 9, column: 3 }, 3),
+            ],
+        ],
+        [
+            "without Rule: description - Scenario",
+            `Feature: a feature file
+  Some feature description
+  is here
+  Scenario: a scenario`,
+            [generateProblem({ line: 4, column: 3 }, 0)],
+        ],
+        [
+            "without Rule: Scenario tags",
+            `Feature: a feature file
+
+
+  @tag1 @tag2
+
+  Scenario: a scenario`,
+            [generateProblem({ line: 6, column: 3 }, 3)],
+        ],
+        [
+            "without Rule: Scenario multiline tags",
+            `Feature: a feature file
+
+  @tag1
+
+  @tag2
+  Scenario Outline: a scenario outline`,
+            [generateProblem({ line: 6, column: 3 }, 2)],
+        ],
+        [
+            "without Rule: Scenario comments",
+            `Feature: a feature file
+  #comment line
+
+
+  Scenario: a scenario`,
+            [generateProblem({ line: 5, column: 3 }, 2)],
+        ],
+        [
+            "without Rule: description - Scenario tags and comments",
+            `Feature: a feature file
+  feture description here
+
+  @tag1
+
+  #comment line
+
+  @tag2
+  Scenario: a scenario`,
+            [generateProblem({ line: 9, column: 3 }, 3)],
+        ],
+        [
+            "without Rule: Background - Scenario tags and comments",
+            `Feature: a feature file
+  feture description here
+
+  Background: a background
+    describe a background
+  @tag1
+  #comment line
+  @tag2
+  Scenario: a scenario`,
+            [generateProblem({ line: 9, column: 3 }, 0)],
+        ],
+        [
+            "Step datatable",
+            `Feature: a feature file
+  Background: a background
+    Given a step
+      | data1 |
+      | data2 |
+  Scenario: a scenario`,
+            [generateProblem({ line: 6, column: 3 }, 0)],
+        ],
+        [
+            "Step docstring",
+            `Feature: a feature file
+  Background: a background
+    Given a step
+      """
+      multiline
+      content
+      """
+
+
+  Scenario: a scenario`,
+            [generateProblem({ line: 10, column: 3 }, 2)],
+        ],
+        [
+            "Examples table",
+            `Feature: a feature file
```

Comment:
There could also be a test case for a feature file that directly starts with:
`Scenario: a scenario`

In that case, the logic has to try and "count" the zero lines before the Scenario starts. And it might crash - we want to avoid a crash, and be able to nicely report all the problems (no Feature line, no newline before Scenario)

Reply:
the rule will return early if there is no Feature keyword i.e. `Scenario: a scenario` (results into invalid feature file)

Reply:
So, crash is handled and invalid feature is reported when we try to lint the following feature file:
```feature
Scenario: a scenario
  Given a step
```